packages/framework/tests/Unit/PostAuthorTest.php

```diff
@@ -203,57 +258,107 @@ public function testAllMethodReturnsEmptyCollectionIfNoAuthorsAreSetInConfig()
     public function testAllMethodReturnsCollectionWithAllAuthorsDefinedInConfig()
     {
         Config::set('hyde.authors', [
-            Author::create('foo'),
+            'foo' => Author::create(),
         ]);
 
         $authors = PostAuthor::all();
 
         $this->assertInstanceOf(Collection::class, $authors);
         $this->assertCount(1, $authors);
-        $this->assertEquals(Author::create('foo'), $authors->first());
+        $this->assertEquals(new PostAuthor('foo'), $authors->first());
     }
 
     public function testMultipleAuthorsCanBeDefinedInConfig()
     {
         Config::set('hyde.authors', [
-            Author::create('foo'),
-            Author::create('bar'),
+            'foo' => Author::create(),
+            'bar' => Author::create(),
         ]);
 
         $authors = PostAuthor::all();
 
         $this->assertInstanceOf(Collection::class, $authors);
         $this->assertCount(2, $authors);
-        $this->assertEquals(Author::create('foo'), $authors->first());
-        $this->assertEquals(Author::create('bar'), $authors->last());
+        $this->assertEquals(new PostAuthor('foo'), $authors->first());
+        $this->assertEquals(new PostAuthor('bar'), $authors->last());
     }
 
     public function testGetMethodReturnsConfigDefinedAuthorByUsername()
     {
         Config::set('hyde.authors', [
-            Author::create('foo', 'bar'),
+            'foo' => Author::create('bar'),
         ]);
+
         $author = PostAuthor::get('foo');
 
         $this->assertInstanceOf(PostAuthor::class, $author);
         $this->assertSame('foo', $author->username);
         $this->assertSame('bar', $author->name);
     }
 
-    public function testGetMethodReturnsNewAuthorIfUsernameNotFoundInConfig()
+    public function testGetMethodNormalizesUsernamesForRetrieval()
+    {
+        Config::set('hyde.authors', [
+            'foo_bar' => Author::create(),
```

Comment:
Should we normalize 'foo bar' to 'foo_bar' when loading the configuration?

However, this might not be necessary because it should match the array key, and normalizing that could be unexpected.

Reply:
Claude:

Consider normalizing usernames in the `Config::set()` call:

The current implementation uses `foo_bar` as the key in the `hyde.authors` configuration. However, the test suggests that various formats (e.g., 'foo bar', 'Foo Bar', 'FOO BAR') should be accepted. 

There are two potential approaches to address this:

1. Normalize the config key:
   Pros: Consistent with retrieval behavior.
   Cons: May lead to unexpected behavior if users rely on exact key matching.

2. Keep the config key as-is and normalize only during retrieval:
   Pros: Maintains expected behavior for config keys.
   Cons: Introduces a discrepancy between config keys and retrieval methods.

Recommendation: Clarify the intended behavior and document the chosen approach. If normalization is desired, consider implementing it consistently across both configuration and retrieval.

Additionally, update the test description to reflect the final decision on normalization behavior.

Reply:
I think after considering the Claude notes, we may want to go with normalization across the board. We normalize for a reason, and it's weird to support the user going against that. If they know what they are doing, they can modify the Author class.

Reply:
A normalization would mean that the following results in a single author:

```php
Config::set('hyde.authors', [
    'foo_bar' => Author::create(),
    'foo-bar' => Author::create(),
    'foo bar' => Author::create(),
    'Foo Bar' => Author::create(),
    'FOO BAR' => Author::create(),
]);
```

But I cannot come up with a reason where such a thing would be desired.

Reply:
**Claude**

Let's think through this:

In most real-world scenarios, it's highly unlikely that a user would intentionally want to create multiple distinct authors with names that differ only in formatting. This kind of duplication would typically be considered undesirable for several reasons:

1. Confusion: It would be confusing for content creators and readers to have multiple author profiles that appear to represent the same person.

2. Inconsistency: It could lead to inconsistent attribution of content, where the same person's work might be associated with different author profiles.

3. Data management: It would complicate data management and queries, potentially leading to incomplete or inaccurate results when trying to gather all content by a specific author.

4. SEO considerations: For websites, having multiple author pages for what is essentially the same person could dilute SEO value and potentially be seen as duplicate content.

However, there might be some edge cases where similar names could be legitimately different:

1. Localization: In some cultures, name variants might be more significant. For example, "Björn" and "Bjorn" might be considered distinct names in some contexts.

2. Pseudonyms: An author might use slightly different pen names for different genres or publications.

3. Historical figures: In academic or historical contexts, there might be a need to distinguish between different spellings of a name used in different periods or documents.

Given these considerations, it seems that normalizing usernames would be the most sensible approach for the vast majority of use cases. It would prevent accidental duplication and ensure consistency.

If you do implement normalization, it would be wise to:

1. Document the normalization rules clearly.
2. Provide a way to override normalization in the rare cases where distinct, similarly-named authors are needed.
3. Implement normalization consistently across all parts of the system (configuration, retrieval, display, etc.).

Reply:
We could add a validator that simply checks the count of the parsed authors and the raw array and warns if they do not match.

Reply:
Implemented in https://github.com/hydephp/develop/pull/1798/commits/d9ceeefc3b9b94e21eae40c8b3f4fac6e839be99